crypto-common/src/lib.rs

```diff
@@ -309,3 +309,34 @@ impl fmt::Display for InvalidLength {
 
 #[cfg(feature = "std")]
 impl std::error::Error for InvalidLength {}
+
+/// Serialized internal state.
+pub type SerializedState<T> = GenericArray<u8, <T as SerializableState>::SerializedStateSize>;
+
+/// The error type returned when an object cannot be deserialized from the state.
+#[derive(Copy, Clone, Eq, PartialEq, Debug)]
+pub struct DeserializeStateError;
+
+impl fmt::Display for DeserializeStateError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> Result<(), fmt::Error> {
+        f.write_str("Deserialization error")
+    }
+}
+
+#[cfg(feature = "std")]
+impl std::error::Error for DeserializeStateError {}
+
+/// Types which can serialize the internal state and be restored from it.
```

Comment:
Do not forget to add the warning that serialized state may contain sensitive data in plain text.